components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/MetricsParams.kt

```diff
@@ -0,0 +1,21 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+ 
+package mozilla.appservices.fxaclient
+
+data class MetricsParams(
+    val parameters: Map<String, String>
```

Comment:
Wasn't sure about my choice here. I first tried to list out the expected parameters:

 - flow_id
 - flow_begin_time
 - device_id
 - utm_source
 - utm_content
 - utm_medium
 - utm_term
  - utm_campaign
  - entrypoint_experiment
  - entrypoint_variation

But it added quite a bit of bloat (Had to null check them since they're all optional and that just explodes up)

But I do get the concern that this does not tell the consumer exactly what the parameters should be.

Reply:
Hrm, yeah, I see what you mean about the amount of boilerplate required. Listing them out individually still feels like the right thing to me, but I'd be curious what other folks think.